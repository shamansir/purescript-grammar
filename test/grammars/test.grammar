main :- [[a-z], repSep(alphaNum, ""), " = 25"].
block :- ["main", ws, blockBody].
blockBody :- ["{", ws, repSep(instr,';'), ws, "}"].
alpha :- ([a-z] | [A-Z] | "_").
instr :- [ident, " = 25"].
num :- [0-9].
alphaNum :- (alpha | num).
ident :- [alpha, repSep((alphaNum | "."), "")].
label :- ident.
ws :- repSep([spaces, (comment | "")], "\n").
spaces :- repSep(" ", "").
comment :- ["//", repSep(commentChar, "")].
commentChar :- ^'\n'.
valueInstr :- [([ident, ws, "=", ws] | ""), rvalue].
rvalue :- (call | const).
const :- int.
int :- [[0-9], repSep([0-9], "")].