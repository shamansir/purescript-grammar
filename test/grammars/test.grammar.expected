main :- repSep(block,ws).
alpha :- ([a-z]|[A-Z]|"_").
alphaNum :- (alpha|num).
block :- ["main",ws,blockBody].
blockBody :- ["{",ws,repSep(instr,';'),ws,"}"].
comment :- ["//",repSep(commentChar,"")].
commentChar :- ^'\n'.
const :- int.
ident :- [alpha,repSep((alphaNum|"."),"")].
instr :- valueInstr.
int :- [[0-9],repSep([0-9],"")].
label :- ident.
num :- [0-9].
rvalue :- (call|const).
spaces :- repSep(" ","").
valueInstr :- [([ident,ws,"=",ws]|""),rvalue].
ws :- repSep([spaces,(comment|"")],"\n").